src/Automation/SnapshotCommand.cs

```diff
@@ -28,40 +28,46 @@ public static IReadOnlyCollection<ScanResults> Execute(Config config, IScanTools
             if (scanTools == null) throw new ArgumentNullException(nameof(scanTools));
             if (scanTools.TargetElementLocator == null) throw new ArgumentException(ErrorMessages.ScanToolsTargetElementLocatorNull, nameof(scanTools));
             if (scanTools.Actions == null) throw new ArgumentException(ErrorMessages.ScanToolsActionsNull, nameof(scanTools));
-            if (scanTools.NativeMethods == null) throw new ArgumentException(ErrorMessages.ScanToolsNativeMethodsNull, nameof(scanTools));
+            if (scanTools.DpiAwareness == null) throw new ArgumentException(ErrorMessages.ScanToolsDpiAwarenessNull, nameof(scanTools));
 
             // We must turn on DPI awareness so we get physical, not logical, UIA element bounding rectangles
-            scanTools.NativeMethods.SetProcessDPIAware();
-
-            if (config.CustomUIAConfigPath != null)
-                scanTools.Actions.RegisterCustomUIAPropertiesFromConfig(config.CustomUIAConfigPath);
-
-            List<ScanResults> resultList = new List<ScanResults>();
-
-            var rootElements = scanTools.TargetElementLocator.LocateRootElements(config.ProcessId);
-
-            if (rootElements is null || !rootElements.Any())
+            object dpiAwarenessObject = scanTools.DpiAwareness.Enable();
```

Comment:
If the call below to scanTools.Actions.Scan() will create parallel tasks, I wonder if IDPIAwareness.Enable() needs to be called in each task since they run in their own threads. That depends where window coordinates are used. My implementation of IDPIAwareness would set the current thread's DPI awareness.

Reply:
I hadn't thought about the multi-threaded case, but it makes total sense. I'll move the DPI calls closer to the code that will eventually be moved into the ThreadProc when async calls are added

Reply:
Thanks!